File: src/main/java/io/github/ytg1234/manhunt/command/ClearCacheCommand.java
package io.github.ytg1234.manhunt.command;

import com.mojang.brigadier.CommandDispatcher;
import io.github.ytg1234.manhunt.ManhuntUtils;
import net.fabricmc.fabric.api.networking.v1.PacketByteBufs;
import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
import net.minecraft.server.command.ServerCommandSource;

import static net.minecraft.server.command.CommandManager.literal;

public class ClearCacheCommand {
    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(literal("clearManhuntCache").executes(context -> {
            ManhuntUtils.haveMod.clear();
            context.getSource().getMinecraftServer().getPlayerManager().getPlayerList().forEach(player -> {
                ServerPlayNetworking.send(player, ManhuntUtils.SERVER_QUESTION_PACKET_ID, PacketByteBufs.empty());
            });
            return 1;
        }));
    }
}

File: manhunt-base/src/main/kotlin/io/github/ytg1234/manhunt/base/command/ClearCacheCommand.kt
package io.github.ytg1234.manhunt.base.command

import com.mojang.brigadier.Command
import com.mojang.brigadier.CommandDispatcher
import io.github.ytg1234.manhunt.base.SERVER_QUESTION_PACKET_ID
import io.github.ytg1234.manhunt.base.haveMod
import mc.aegis.AegisCommandBuilder
import net.fabricmc.fabric.api.networking.v1.PacketByteBufs
import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking
import net.minecraft.server.command.ServerCommandSource

object ClearCacheCommand {
    fun register(dispatcher: CommandDispatcher<ServerCommandSource>) {
        dispatcher.register(
            AegisCommandBuilder("clearManhuntCache") {
                requires { it.hasPermissionLevel(2) }
                executes { ctx ->
                    haveMod.clear()
                    ctx.source.minecraftServer.playerManager.playerList.forEach {
                        ServerPlayNetworking.send(it, SERVER_QUESTION_PACKET_ID, PacketByteBufs.empty())
                    }
                    Command.SINGLE_SUCCESS
                }
            }.build()
        )
    }
}
